### PR TITLE
os: add method `tmpdir()`

### PR DESCRIPTION
**Pull request by yorkie**
Implements the Node.js `os` API: https://nodejs.org/dist/latest-v12.x/docs/api/os.html#os_os_tmpdir

##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added

### CODE REVIEW

**Comment by yorkie** [comment]
Note: this is a prepared PR for adding the `fsync` and `fdatasync()` :)

File: src/js/os.js [hunk 1]
```diff
@@ -88,3 +88,18 @@ Object.defineProperty(exports, 'EOL', {
...
   }
 });
+
+function isStringEndsWith(str, search) {
+  var len = str.length;
+  return str.substring(len - search.length, len) === search;
+}
+
+exports.tmpdir = function tmpdir() {
+  var path = process.env.TMPDIR ||
```
Comment (lolBig) [inline src/js/os.js]: why not get env from [getenv(3)](http://man7.org/linux/man-pages/man3/getenv.3.html)

Comment (yorkie) [reply src/js/os.js]: What's the benefits in that way? Actually we did set the env at [`GetEnvironArray`](https://github.com/yodaos-project/ShadowNode/blob/master/src/modules/iotjs_module_process.c#L423) via [`environ(7)`](http://man7.org/linux/man-pages/man7/environ.7.html).

**Comment by yorkie** [comment]
Restarts the failed job(https://travis-ci.com/yodaos-project/ShadowNode/jobs/246598500) caused by networks.

File: test/run_pass/test_os.js [hunk 2]
```diff
@@ -35,3 +35,21 @@ var expected = [
 ];
 
 assert.deepStrictEqual(actual, expected);
+
+// test for tmpdir
```
Comment (legendecas) [inline test/run_pass/test_os.js]: A default case shall be covered. i.e. no `TMPDIR` nor `TMP` and `TEMP` environs are set.

File: src/js/os.js (hunk 1 above)
Comment (lolBig) [reply src/js/os.js]: in Windows, Node.js is also in this way，but on other [platforms](https://github.com/nodejs/node/blob/908292cf1f551c614a733d858528ffb13fb3a524/lib/os.js#L133) are via [getenv](https://github.com/nodejs/node/blob/a8277ddb5505056e3d66d02051c9195310c5c715/src/node_credentials.cc#L61), i am not sure why Node.js is doing this.

Comment (legendecas) [reply src/js/os.js]: getenv is only used if node environment has not been setup.

File: docs/api/OS.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ The properties available on the assigned network address object include:
 * `scopeid` {Number} The numeric IPv6 scope ID (only specified when family is IPv6)
 * `cidr` {String} The assigned IPv4 or IPv6 address with the routing prefix in CIDR notation. If the netmask is invalid, this property is set to null
 
-```js
```
[the remark this answers is not in the document]
Comment (yorkie) [reply docs/api/OS.md]: Yea, the below lines actually are not in JavaScript.

File: src/js/os.js (hunk 1 above)
Comment (yorkie) [reply src/js/os.js]: See https://github.com/nodejs/node/pull/18511 for all related history, this seems a historic design problem that allows packages sharing via `process.env`(write/read), but anyway, we have to handle this security, thanks @lolBig

File: test/run_pass/test_os.js (hunk 2 above)
Comment (yorkie) [reply test/run_pass/test_os.js]: Ok, BTW I just copied these tests from here: https://github.com/nodejs/node/blob/master/test/parallel/test-os.js#L46-L78.

**Pull request closed without merging**